decidim-core/app/models/decidim/user.rb

```diff
@@ -106,7 +106,7 @@ def send_devise_notification(notification, *args)
     end
 
     def all_roles_are_valid
-      errors.add(:roles, :invalid) unless roles.all? { |role| ROLES.include?(role) }
+      errors.add(:roles, :invalid) unless roles.compact.all? { |role| ROLES.include?(role) }
```

Comment:
I think this is too defensive - shouldn't be needed if we take care when inserting roles, right?

Reply:
This was to avoid running a migration changing those values, what do you prefer?

Reply:
Well I guess it's fine the way it is then.